pkg/k8s/watchers/watcher.go

```diff
@@ -443,8 +445,8 @@ func (k *K8sWatcher) EnableK8sWatcher(ctx context.Context, resources []string) e
 		case k8sAPIGroupCiliumClusterwideNetworkPolicyV2:
 			k.ciliumClusterwideNetworkPoliciesInit(ciliumNPClient)
 		case k8sAPIGroupCiliumEndpointV2:
-			asyncControllers.Add(1)
-			go k.ciliumEndpointsInit(ciliumNPClient, asyncControllers)
+		case k8sAPIGroupCiliumEndpointSliceV2Alpha1:
+			k.initCiliumEndpointOrSlices(ciliumNPClient, asyncControllers)
```

Comment:
This needs to occur in the `k8sAPIGroupCiliumEndpointV2` case, then the `k8sAPIGroupCiliumEndpointSliceV2Alpha1` case should explicitly mention that it's a no-op that relies on  `k8sAPIGroupCiliumEndpointV2`, like the other cases above. This will ensure that the initialization occurs for the correct type regardless of whether you turn this feature on or not, and will also help to describe this for the next reader of this code.

I believe that this is the likely source of your CI failures.

Reply:
I think you're right. That's what is happening. Let me test the code and update PR.

Reply:
Updated PR.